src/Xamarin.Android.Build.Tasks/Utilities/ZipArchiveEx.cs

```diff
@@ -52,18 +58,23 @@ string ArchiveNameForFile (string filename, string directoryPathInZip)
 			return pathName.Replace ("\\", "/");
 		}
 
+		string TrimEntryName (string entryName, bool appendSeparator = false)
+		{
+			return entryName.Replace ("\\", "/").TrimStart ('/') + (appendSeparator ? "/" : "");
+		}
```

Comment:
Could this be done in `ArchiveNameForFile`? Just so it's all in the same place.

Reply:
not really, I don't think. its also used here https://github.com/xamarin/xamarin-android/pull/5266/files#diff-d2b91e191cec60e8b907d7d6ddb09d24e4ad619c57d7146537af332a13bfa4faR116. That logic if it was passed though `ArchiveNameForFile` would change the resulting directory.

I mean I can try it and see what breaks.